### PR TITLE
Added utility to expand list of numbers with repeats

### PR DESCRIPTION
**Pull request by obi-isebor**

### CODE REVIEW

File: ResSimpy/Utils/general_utilities.py [hunk 1]
```diff
@@ -12,3 +15,43 @@ def check_if_string_is_float(x: str) -> bool:
...
+
+    Returns:
+        int | float: integer or float value from input numerical string
+    """
+    try:
+        # Try to convert the string to an integer
+        return int(x)
+    except ValueError:
+        # If it fails, try to convert the string to a float
+        return float(x)
```
Comment (jonnyh404) [inline ResSimpy/Utils/general_utilities.py:34]: When this fails i.e. passed a Non-numeric string the error message will be very poor - might be worth thinking about how to expand the error message to include more detail?

File: tests/test_utils.py [hunk 2]
```diff
@@ -204,3 +206,15 @@ def test_to_string_generic(mocker, headers, expected_result):
 
     # Assert
     assert result_string == expected_result
+
+
+def test_expand_string_list_of_numbers():
```
Comment (jonnyh404) [inline tests/test_utils.py:211]: Please add a test where nothing gets changed in the string.

**Review by cflynn3 — APPROVED** [review]
Aside from Jonny's suggestions I think this looks good.

**Review by jonnyh404 — APPROVED** [review]
Looks good to me.

**Pull request merged**